packages/utils/src/serialize.js

```diff
@@ -1,7 +1,36 @@
 import serialize from 'serialize-javascript'
 
+function normalizeFunctions(obj) {
+  if (typeof obj !== 'object' || Array.isArray(obj) || obj === null) {
+    return obj
+  }
+  for (const key in obj) {
+    if (key === '__proto__' || key === 'constructor') {
+      continue
+    }
+    const val = obj[key]
+    if (val !== null && typeof val === 'object' && !Array.isArray(obj)) {
+      obj[key] = normalizeFunctions(val)
+    }
+    if (typeof obj[key] === 'function') {
+      const asString = obj[key].toString()
+      const match = asString.match(/^([^{(]+)=>\s*(.*)/)
+      if (match) {
+        let functionBody = match[2].match(/^{?(\s*return\s+)?(.*?)}?$/)[2].trim()
```

Comment:
@pimlie good catch, pushing a fix in a bit.

Reply:
Actually it's fine, I've updated the test suite to demonstrate that.

Reply:
The tests are unit testing `serializeFunction`, not `normalizeFunction`. If you would unit test normalieFunction it would fail

```js
const arr = [1]
const fn = new Function('arg', '2*arg')
console.log(arr.map(fn))
```

That logs `[undefined]` and not `[2]`

I had the same issue with `tib` but I decided to use babel for parsing as it felt like a much more stable solution then hacking some regexes together ;)

Reply:
I see what you mean, I'll double check it. Also, can you point me to where you use babel similarly in tib? :)

Reply:
Its here: https://github.com/pimlie/tib/blob/master/src/utils/index.js#L49

Not sure if it can be easily used here as I also use transform to turn the AST back to code

Reply:
@pimlie can you take a look again?

Reply:
The tests look solid to me, so the implementation should be good as well! Thanks :)